### ISSUE TITLE
Support asynchronous requests (accept `aiohttp.ClientSession` in `berserk.Client`?)

### ISSUE THREAD

**Issue by jacksonthall22** [lichess-org/berserk#18]
I'm writing a discord bot in PyCord, where [the docs suggest](https://docs.pycord.dev/en/stable/faq.html#what-does-blocking-mean) performing all requests asynchronously. To be able to use `berserk` for my bot I would love if something like this worked out of the box:
```python
async with aiohttp.ClientSession() as session:
    client = berserk.Client(session=session)
    ...  # Make api calls etc.
```

**Comment by icp1994** [comment]
https://lichess.org/api#section/Introduction/Rate-limiting mentions: "Only make one request at a time."

**Comment by benediktwerner** [comment]
Yeah, there probably are not a lot of cases where async is really beneficial when using the Lichess API. Though it might still be useful to handle timeouts and stuff like that nicer or when already using async for other things.

It looks like it should theoretically be possible to transparently use an async client but getting streams and types to work properly might be a bit tricky. But if anybody can find a nice solution, I'd be open to merging it.
